src/libcrun/cgroup.c

```diff
@@ -2082,19 +2082,22 @@ write_memory_resources (int dirfd, bool cgroup2, runtime_spec_schema_config_linu
       if (UNLIKELY (ret < 0))
         return ret;
     }
-  if (memory->swap)
+  if (memory->swap_present)
     {
       char swap_buf[32];
       size_t swap_buf_len;
 
+      if (cgroup2 && memory->swap_present && !memory->limit_present)
```

Comment:
Do we have to worry about the other way around?
if (cgroup2 && !memory->swap_present && memory->limit_present)

Reply:
the memory limit can be written directly without being in relation with the swap.

@kolyshkin does it fix the issue you've seen?

Reply:
Tested with the second commit applied, works as intended, so yes, it fixes #315

```console
[root@f31-test tst]# tail /sys/fs/cgroup/xex/mem*max 
==> /sys/fs/cgroup/xex/memory.max <==
536870912

==> /sys/fs/cgroup/xex/memory.swap.max <==
0
[root@f31-test tst]# ~kir/bin/crun update --memory=536870912 --memory-swap=$((536870912*2)) xex
[root@f31-test tst]# tail /sys/fs/cgroup/xex/mem*max 
==> /sys/fs/cgroup/xex/memory.max <==
536870912

==> /sys/fs/cgroup/xex/memory.swap.max <==
536870912
[root@f31-test tst]# ~kir/bin/crun update --memory=536870912 --memory-swap=$((536870912*3)) xex
[root@f31-test tst]# tail /sys/fs/cgroup/xex/mem*max 
==> /sys/fs/cgroup/xex/memory.max <==
536870912

==> /sys/fs/cgroup/xex/memory.swap.max <==
1073741824
[root@f31-test tst]# ~kir/bin/crun update --memory-swap=$((536870912*3)) xex
2020-04-07T21:02:25.000955454Z: cannot set swap limit without the memory limit
[root@f31-test tst]# tail /sys/fs/cgroup/xex/mem*max 
==> /sys/fs/cgroup/xex/memory.max <==
536870912

==> /sys/fs/cgroup/xex/memory.swap.max <==
1073741824
```